python/pyarrow/tests/parquet/test_metadata.py

```diff
@@ -433,7 +434,9 @@ def test_write_metadata(tempdir):
     assert parquet_meta_mult.num_row_groups == 2
 
     # append metadata with different schema raises an error
-    with pytest.raises(RuntimeError, match="requires equal schemas"):
+    msg = ("AppendRowGroups requires equal schemas.\n"
+           "These two columns differ at index: 0")
```

Comment:
The message is a bit confusing and it took me some time to understand what "differ at index 0" meant.
I would suggest "The two columns with index 0 differ".